### ISSUE TITLE
Data format parsing crash if values is empty array.

### ISSUE THREAD

**Issue by Lacroute** [vega/vega#1036]
Hi,

You can try this spec and see the bug (based on the bar chart exemple):
```
{
  "$schema": "https://vega.github.io/schema/vega/v3.0.json",
  "width": 400,
  "height": 200,
  "padding": 5,

  "data": [
    {
      "name": "table",
      "values": [],
      "format": {
        "parse": {"date": "date:%Y-%m-%d"}
      }
    }
  ],

  "signals": [
    {
      "name": "tooltip",
      "value": {},
      "on": [
        {"events": "rect:mouseover", "update": "datum"},
        {"events": "rect:mouseout",  "update": "{}"}
      ]
    }
  ],

  "scales": [
    {
      "name": "xscale",
      "type": "band",
      "domain": {"data": "table", "field": "category"},
      "range": "width",
      "padding": 0.05,
      "round": true
    },
    {
      "name": "yscale",
      "domain": {"data": "table", "field": "amount"},
      "nice": true,
      "range": "height"
    }
  ],

  "axes": [
    { "orient": "bottom", "scale": "xscale" },
    { "orient": "left", "scale": "yscale" }
  ],

  "marks": [
    {
      "type": "rect",
      "from": {"data":"table"},
      "encode": {
        "enter": {
          "x": {"scale": "xscale", "field": "category"},
          "width": {"scale": "xscale", "band": 1},
          "y": {"scale": "yscale", "field": "amount"},
          "y2": {"scale": "yscale", "value": 0}
        },
        "update": {
          "fill": {"value": "steelblue"}
        },
        "hover": {
          "fill": {"value": "red"}
        }
      }
    },
    {
      "type": "text",
      "encode": {
        "enter": {
          "align": {"value": "center"},
          "baseline": {"value": "bottom"},
          "fill": {"value": "#333"}
        },
        "update": {
          "x": {"scale": "xscale", "signal": "tooltip.category", "band": 0.5},
          "y": {"scale": "yscale", "signal": "tooltip.amount", "offset": -2},
          "text": {"signal": "tooltip.amount"},
          "fillOpacity": [
            {"test": "datum === tooltip", "value": 0},
            {"value": 1}
          ]
        }
      }
    }
  ]
}
```

The fix is to have the `table` `values` (line 10) to `[{}]`. It's related to the format parsing, if you remove the `format` entry (line 11 - 12 - 13) it works again.

_Found by Roy in https://groups.google.com/forum/#!topic/vega-js/CtMcKWICewE_

**Comment by jheer** [comment]
Thanks for sharing the issue! This is indeed a bug; we'll investigate.

**Comment by jheer** [comment]
Fixed in [vega-loader v2.0.2](https://github.com/vega/vega-loader/releases/tag/v2.0.2). Will be included in the next Vega release.
